snuba/query/parser/strings.py

```diff
@@ -42,12 +42,12 @@ def parse_string_to_expr(val: str) -> Expression:
     # TODO: This will use the schema of the dataset/entity to decide if the expression is
     # a column or a literal.
     if val.isdigit():
-        return Literal(val, int(val))
+        return Literal(None, int(val))
     else:
         try:
-            return Literal(val, float(val))
+            return Literal(None, float(val))
         except Exception:
             if QUOTED_LITERAL_RE.match(val):
-                return Literal(val[1:-1], val[1:-1])
```

Comment:
I think this was actually a mistake. Applying aliases by default to literals is a bad idea. It can cause shadowing very easily and they cannot be safely removed during formatting because each type of literal is escaped differently. 
Also they are unnecessary because our language does not allow top level literals in the select clause, so no risk of loosing the alias during transformation. If we allowed that in the language we would need to take a different approach to parsing.